src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -1081,13 +1081,34 @@ static String getRanges(JsonArray ar, String startProp, String endProp) {
         });
   }
 
-  /** Draft, Work in Progress (WIP). */
   Future<Void> getUseOverTime(Vertx vertx, RoutingContext ctx) {
+    String format = ctx.request().params().get("format");
+    switch (format) {
+      case "JOURNAL": return getUseOverTimeJournal(vertx, ctx);
+      case "BOOK": return getUseOverTimeBook(vertx, ctx);
+      case "DATABASE": return getUseOverTimeDatabase(vertx, ctx);
+      default: throw new IllegalArgumentException("format = " + format);
+    }
+  }
+
+  Future<Void> getUseOverTimeJournal(Vertx vertx, RoutingContext ctx) {
     TenantPgPool pool = TenantPgPool.pool(vertx, TenantUtil.tenant(ctx));
     String agreementId = ctx.request().params().get("agreementId");
     String start = ctx.request().params().get("startDate");
     String end = ctx.request().params().get("endDate");
 
+    return getUseOverTimeJournal(pool, agreementId, start, end)
+        .map(json -> {
+          ctx.response().setStatusCode(200);
+          ctx.response().putHeader("Content-Type", "application/json");
+          ctx.response().end(json.encodePrettily());
+          return null;
+        });
+  }
+
+  Future<JsonObject> getUseOverTimeJournal(TenantPgPool pool,
+      String agreementId, String start, String end) {
+
     LocalDate startDate = LocalDate.parse(start).withDayOfMonth(1);
```

Comment:
I don't know the library you're using here, but I hope its not interpreting the start- and end-date parameters as locale-appropriate formatted strings? We send ISO-format dates.

Reply:
A date without a time-zone in the ISO-8601 calendar system, such as 2007-12-03.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDate.html

Reply:
They don't do the same thing. One of them creates an HTTP response with content-type and whatnot.